lib/bolt/inventory.rb

```diff
@@ -0,0 +1,101 @@
+require 'set'
+require 'bolt/util'
+require 'bolt/target'
+require 'bolt/inventory/group'
+
+module Bolt
+  class Inventory
+    class ValidationError < Bolt::Error
+      attr_accessor :path
+      def initialize(message, offending_group)
+        super(msg, 'bolt.inventory/validation-error')
+        @_message = message
+        @path = offending_group ? [offending_group] : []
+      end
+
+      def details
+        { 'path' => path }
+      end
+
+      def add_parent(parent_group)
+        @path << parent_group
+      end
+
+      def message
+        "#{@_message} for group at #{path}"
+      end
+    end
+
+    def self.default_paths
+      [File.expand_path(File.join('~', '.puppetlabs', 'bolt', 'inventory.yaml'))]
+    end
+
+    def self.from_config(config)
+      data = Bolt::Util.read_config_file(config[:inventoryfile], default_paths)
```

Comment:
We should pass `'inventory'` at the end here, or it will log `Could not read file file` if there's a problem.